sensor/common/networkflow/manager/manager_impl.go

```diff
@@ -506,35 +507,85 @@ func (m *networkFlowManager) enrichAndSendProcesses() {
+	}
+	return failReasons
+}
+
+func formatMultiErrorOneline(errs []error) string {
+	elems := make([]string, len(errs))
+	for i, err := range errs {
+		// The error is used in debug logs and is much nicer to read with the first letter capitalized.
+		// Writing the error message with first capital in the code raises a style warning.
+		msg := cases.Title(language.English, cases.NoLower).String(err.Error())
+		elems[i] = fmt.Sprintf("(%d) %s", i+1, msg)
+	}
+	return strings.Join(elems, ", ")
+}
+
+func logReasonForAggregatingNetGraphFlow(conn *connection, contNs, contName, entitiesName string, port uint16, failReason *multierror.Error) {
+	reasonStr := ""
+	if failReason != nil {
+		failReason.ErrorFormat = formatMultiErrorOneline
+		reasonStr = failReason.Error()
+	}
+	// No need to produce complex chain of reasons, if there is one simple explanation
+	if conn.remote.IsConsideredExternal() {
+		reasonStr = "Collector did not report the IP address to Sensor - the remote part is the Internet"
+	}
+	if conn.incoming {
+		// Keep internal wording even if central lacks `NetworkGraphInternalEntitiesSupported` capability.
+		log.Debugf("Marking incoming connection to container %s/%s from %s:%s as '%s' in the network graph: %s.",
+			contNs, contName, conn.remote.IPAndPort.String(),
+			strconv.Itoa(int(port)), entitiesName, reasonStr)
+	} else {
+		log.Debugf("Marking outgoing connection from container %s/%s to %s as '%s' in the network graph: %s.",
+			contNs, contName, conn.remote.IPAndPort.String(),
+			entitiesName, reasonStr)
+	}
+}
+
 func (m *networkFlowManager) enrichConnection(conn *connection, status *connStatus, enrichedConnections map[networkConnIndicator]timestamp.MicroTS) {
 	timeElapsedSinceFirstSeen := timestamp.Now().ElapsedSince(status.firstSeen)
 	isFresh := timeElapsedSinceFirstSeen < clusterEntityResolutionWaitPeriod
+	var netGraphFailReason *multierror.Error
 
 	container, ok := m.clusterEntities.LookupByContainerID(conn.containerID)
 	if !ok {
-		// Expire the connection if the container cannot be found within the clusterEntityResolutionWaitPeriod
-		if timeElapsedSinceFirstSeen > maxContainerResolutionWaitPeriod {
-			if activeConn, found := m.activeConnections[*conn]; found {
-				enrichedConnections[*activeConn] = timestamp.Now()
-				delete(m.activeConnections, *conn)
-				flowMetrics.SetActiveFlowsTotalGauge(len(m.activeConnections))
-				return
-			}
-			status.rotten = true
-			// Only increment metric once the connection is marked rotten
-			flowMetrics.ContainerIDMisses.Inc()
-			log.Debugf("Unable to fetch deployment information for container %s: no deployment found", conn.containerID)
-		}
+		// There is an incoming connection to a container that we do not know.
+		// 90% of the cases that container is Sensor itself before being restarted.
+		err := m.handleContainerNotFound(conn, status, enrichedConnections)
+		log.Debugf("Enrichment failed: %v", err)
```

Comment:
[nit]
```suggestion
		if err := m.handleContainerNotFound(conn, status, enrichedConnections); err != nil {
			log.Debugf("Enrichment failed: %v", err)
		}
```

Reply:
Actually, in it's current form it always returns an error, but it makes sense to build that in.

Reply:
Addressed in [354d424](https://github.com/stackrox/stackrox/pull/13463/commits/354d42440fc688c89404ab4815ff498cde0cc862)